File: src/main/java/client/inventory/ModifyInventoryContextInterface.java
package client.inventory;

import client.inventory.item.templates.ItemTemplate;
import client.inventory.slots.ItemSlot;
import client.inventory.slots.ItemSlotBundle;
import util.packet.PacketWriter;

public interface ModifyInventoryContextInterface {

    void add(ItemSlot item);

    void add(ItemTemplate item, short quantity);

    void set(short slot, ItemSlot item);

    void set(short slot, ItemTemplate item, short quantity);

    void remove(short slot);

    void remove(short slot, short quantity);

    void remove(ItemSlot item);

    void remove(ItemSlot item, short quantity);

    void remove(int id, short quantity);

    void move(short from, short to);

    ItemSlotBundle take(short slot, short quantity);

    ItemSlotBundle take(ItemSlotBundle bundle, short quantity);

    ItemSlotBundle take(int id, short quantity);

    void update(short slot);

    void update(ItemSlot item);

    void encode(PacketWriter pw);
}

File: src/main/java/client/inventory/ModifyInventoryContextInterface.kt
package client.inventory

import client.inventory.item.templates.ItemTemplate
import client.inventory.slots.ItemSlot
import client.inventory.slots.ItemSlotBundle
import util.packet.PacketWriter

interface ModifyInventoryContextInterface {
    fun add(item: ItemSlot)
    fun add(item: ItemTemplate, quantity: Short)
    operator fun set(slot: Short, item: ItemSlot)
    operator fun set(slot: Short, item: ItemTemplate, quantity: Short)
    fun remove(slot: Short)
    fun remove(slot: Short, quantity: Short)
    fun remove(item: ItemSlot)
    fun remove(item: ItemSlot, quantity: Short)
    fun remove(id: Int, quantity: Short)
    fun move(from: Short, to: Short)
    fun take(slot: Short, quantity: Short): ItemSlotBundle
    fun take(bundle: ItemSlotBundle, quantity: Short): ItemSlotBundle
    fun take(id: Int, quantity: Short): ItemSlotBundle
    fun update(slot: Short)
    fun update(item: ItemSlot)
    fun encode(pw: PacketWriter)
}